### PR TITLE
Incorrect tests fixed

### PR DESCRIPTION
**Pull request by johanandren**
Some scaladoc boyscouting as bonus

### CODE REVIEW

File: akka-http-tests/src/test/java/docs/http/javadsl/server/HandlerExampleDocTest.java [hunk 1]
```diff
@@ -118,16 +118,16 @@ Route createRoute() {
...
     r.run(HttpRequest.GET("/calculator/path-multiply/23/5"))
-     .assertStatusCode(200)
-     .assertEntity("x * y = 115");
+      .assertStatusCode(StatusCodes.OK)
+      .assertEntity("x * y = 115");
 
-    r.run(HttpRequest.GET("/calculator/subtract?x=42&y=12"))
-     .assertStatusCode(200)
-     .assertEntity("x - y = 30");
+    r.run(HttpRequest.GET("/calculator/subtract/42/12"))
```
Comment (ktoso) [inline akka-http-tests/src/test/java/docs/http/javadsl/server/HandlerExampleDocTest.java:128]: ah :) okey

File: akka-http-tests/src/test/java/docs/http/javadsl/server/HandlerExampleDocTest.java [hunk 2]
```diff
@@ -30,16 +30,16 @@ public void testSimpleHandler() {
       //#simple-handler
       Route handlerString = extractMethod(method ->
         extractUri(uri ->
-          complete(String.format("This was a %s request to %s", method, uri))
+          complete(String.format("This was a %s request to %s", method.name(), uri))
```
Comment (ktoso) [inline akka-http-tests/src/test/java/docs/http/javadsl/server/HandlerExampleDocTest.java:33]: could method get a better toString instead?

Comment (johanandren) [inline akka-http-tests/src/test/java/docs/http/javadsl/server/HandlerExampleDocTest.java:33]: Currently toString is "HttpMethod(GET)". Not sure if it is a good idea having it toString to the name. You decide.

Comment (johanandren) [inline akka-http-tests/src/test/java/docs/http/javadsl/server/HandlerExampleDocTest.java:33]: Like in, there is an explicit impl toStringing it like that

Comment (ktoso) [inline akka-http-tests/src/test/java/docs/http/javadsl/server/HandlerExampleDocTest.java:33]: Oh I see. I'll check for consistency with ScalaDSL then

Comment (ktoso) [inline akka-http-tests/src/test/java/docs/http/javadsl/server/HandlerExampleDocTest.java:33]: Ok, it's consistent

**Pull request merged**